### DISCUSSION TITLE
[RFC] Ability to edit certain relations as "SubResource" on a resource edit page

### DISCUSSION THREAD

**Discussion by mallardduck** [filamentphp/filament#319] [Ideas]
## TL;DR:
The gist of the idea would be a mechanism similar to RelationManager that can be registered to a resource.
However this differs as this would provide an on page form for the SubResource.
The main target in mind would be data stored in `HasOne` relationships that simply don't need a dedicated page/resource.

## Description:
The main objective here would be for exposing fields that for _reasons_ are stored as a separate Model, but is conceptually part of the same resource. Put another way - the entities this feature would expose to a resource would not fit a model that requires a UX for attach/detach.

One concrete example of this might be a very simple blog implementation where SEO/open-graph meta-data is stored in a table separate the main blog posts. In this case a Developer may be requested to facilitate the creation of a single page where both the Blog Post and this meta data can be edited. While the RelationshipManager can facilitate this for the most part - this feature would allow devs to create a more seamless user experience.

Ultimately, this could be accomplished via two forms on the same page (i.e. two different save/cancel actions) or by a subform concept with a single set of action buttons. However the former is likely easier than the later - but potentially both mechanisms could be supported alongside each other.

## Future
As mentioned, the initial build could/should probably be simply adding a second form for the sub resource.
This is a good default user experience for the new feature, however this could be expanded later.

A no brainer, would be an option to embed the fields into the main form directly to circumvent two fields.
The other areas of expansion could be around more complex relations that could benefit UX similar to `HasOne`.
So potentially this could be expanded to support `HasOneThrough` as well and other such similar relations.

---

## DB Example:
Note: both tables here are truncated just to show the basic `HasOne` relationship in mind.
### blog_posts:
| Name  | Type |
| ------------- | ------------- |
| ID  | INT increments  |
| TITLE | string  |
| body | text  |

### blog_post_metas:
| Name  | Type |
| ------------- | ------------- |
| ID  | INT increments  |
| blog_post_ID | INT foreign |
| TITLE | string  |
| descriptions | text  |

**Comment by mallardduck** [comment]
Here is a screenshot of the working proof of concept in PR: https://github.com/laravel-filament/filament/pull/318

![Screen Shot 2021-04-04 at 9 49 58 PM](https://user-images.githubusercontent.com/619938/113528837-d9b03d80-958f-11eb-92ed-e58d3f160b43.png)

**Reply by anishsarkar** [reply]
Hi. I need exactly this. But I don't see any documentation or code examples. Could you post some code for this. 

Thanks in advance.
